### PR TITLE
Don't analyse non-SQL cards with Macaw

### PR DESCRIPTION
**Pull request by crisptrutski**
Refs https://github.com/metabase/metabase/issues/43516

### Description

Currently Macaw will try and fail on all non-SQL native cards, e.g. MongoDB. This saves all that blood, sweat, and tears.

The long term solution is to flip this around and call a multi-method which does the analysis, but we're holding back on a new driver method until we've rejigged things to be EE-only and figured out some naming.

### CODE REVIEW

File: src/metabase/native_query_analyzer.clj [hunk 1]
```diff
@@ -138,24 +139,32 @@
...
-          macaw-opts   (nqa.impl/macaw-options driver)
-          parsed-query (macaw/query->components (macaw/parsed-query sql-string) macaw-opts)
-          explicit-ids (explicit-field-ids-for-query parsed-query db-id)
-          implicit-ids (set/difference
-                        (implicit-field-ids-for-query parsed-query db-id)
-                        explicit-ids)]
-      {:explicit   explicit-ids
-       :implicit implicit-ids})))
+  (when (and (active?) (:native query))
+    (let [driver (driver.u/database->driver (:database query))]
```
Comment (calherries) [inline src/metabase/native_query_analyzer.clj:166]: This is not really related to the PR, but should we be using `metabase.lib` to interact with the query? If we were taking modularization seriously (and I know we are breaking it all over the place) we should be using `metabase.lib.native/engine` to get the `driver` for the query

**Review by calherries — APPROVED** [review]
LGTM

File: src/metabase/native_query_analyzer.clj (hunk 1 above)
Comment (crisptrutski) [reply src/metabase/native_query_analyzer.clj:166]: TIL, thanks for pointing this out
